File: cpp/repeating_string/repeating_string.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the repeatedString function below.
long repeatedString(string s, long n) {
    long occurrences = 0;
    long subLen = s.size();
    long l = min(subLen, n);

    for (int i = 0; i < l; i++) {
        if (s[i] == 'a') {
            occurrences++;
        }
    }

    if (n > subLen) {
        occurrences = occurrences * (n / subLen);

        for (int j = 0; j < (n % subLen); j++) {
            if (s[j] == 'a') {
                occurrences++;
            }   
        }
    }

    return occurrences;
}

int main()
{
    string s = "ababa";
    long n = 3;
    long occurrences = repeatedString(s, n);
    cout << occurrences << '\n';
    return 0;
}

File: java/repeating_string/RepeatingString.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class RepeatingString {

    // Complete the repeatedString function below.
    static long repeatedString(String s, long n) {
        long strLen = s.length();
        long l = Math.min(strLen, n);
        long occur = 0;

        for (int j = 0; j < l; j++) {
            if (s.charAt(j) == 'a') {
                occur++;
            }
        }

        if (n > strLen) {
            occur = occur * (n / strLen);
        
            for (int i = 0; i < (n % strLen); i++) {
                if (s.charAt(i) == 'a') {
                    occur++;
                }
            }
        }


        return occur;
    }

    public static void main(String[] args) {
        String s = "ababa";
        long n = 3;
        long occur = repeatedString(s, n);
        System.out.println(occur);
    }
}
